### PR TITLE
Fire tick event

### PR DESCRIPTION
**Pull request by Artem226**
Hello, Minecraft Forge!

I've added a new event - FireTickEvent.

This event allows modders to do something when fire updates.
If this event is cancelled fire won't do anything on update.(Like disabling "doFireTick" rule)

I hope you merge this PR.

### CODE REVIEW

**Comment by ShetiPhian** [comment]
Since your attempting to get a Fire event added, you can roll this one into your pr if you want. 
https://github.com/MinecraftForge/MinecraftForge/pull/2390

**Comment by Artem226** [comment]
@ShetiPhian , No, thanks.

**Comment by Zetal911** [comment]
This would be extremely useful, and as far as I have seen is probably the ideal solution to a "Fire" event.

File: src/main/java/net/minecraftforge/event/world/FireTickEvent.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * This event does not have a result. {@link HasResult}<br>
+ * <br>
+ * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
+ * 
+ * @author Artem226
+ */
+@Cancelable
+public class FireTickEvent extends BlockEvent {
+	
+	public final Random rand;
```
Comment (kashike) [inline src/main/java/net/minecraftforge/event/world/FireTickEvent.java]: If you want to get this pulled into 1.9 as well, make this field `private` and add a `getRandom` accessor:

Comment (Artem226) [inline src/main/java/net/minecraftforge/event/world/FireTickEvent.java]: @kashike I'll add it if necessary.

Comment (kashike) [inline src/main/java/net/minecraftforge/event/world/FireTickEvent.java]: It's necessary to be pulled into 1.9, there are no more public event fields, replaced with accessors.

File: src/main/java/net/minecraftforge/event/world/FireTickEvent.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * <br>
+ * <br>
+ * This event is {@link Cancelable}.<br>
+ * 
+ * <br>
+ * This event does not have a result. {@link HasResult}<br>
+ * <br>
+ * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
+ * 
+ * @author Artem226
```
Comment (williewillus) [inline src/main/java/net/minecraftforge/event/world/FireTickEvent.java]: no

File: src/main/java/net/minecraftforge/common/ForgeHooks.java [hunk 3]
```diff
@@ -937,4 +938,12 @@ public static Vec3 rayTraceEyeHitVec(EntityLivingBase entity, double length)
         MovingObjectPosition movingObjectPosition = rayTraceEyes(entity, length);
         return movingObjectPosition == null ? null : movingObjectPosition.hitVec;
     }
+    
+    public static boolean onFireTick(World worldIn, BlockPos pos, IBlockState state, Random rand)
+    {
+    	FireTickEvent fte = new FireTickEvent(worldIn, pos, state, rand);
```
Comment (Lunatrius) [inline src/main/java/net/minecraftforge/common/ForgeHooks.java]: Mixed spaces/tabs.

File: src/test/java/net/minecraftforge/test/FireTickTest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import net.minecraftforge.fml.common.Mod.EventHandler;
+import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+@Mod(modid="FireTickTest", name="FireTickTest", version="0.0.0")
+public class FireTickTest {
+
+	public static final boolean ENABLE = false;
+	
+	@EventHandler
```
Comment (Lunatrius) [inline src/test/java/net/minecraftforge/test/FireTickTest.java]: Same thing here, `init` is using spaces, the rest of the file is using tabs.

**Comment by immibis** [comment]
Why not a generic random-block-tick event?

**Comment by Artem226** [comment]
@immibis , because it will spam event handlers. There are millions of blocks in Minecraft. If all of them send events, it will cause a terrible performance fall.

A FireTickEvent will be sent only when fire updates.

**Comment by Spartan322** [comment]
@Artem226 I don't think that was the meaning of that suggestion, I think he means make a generic event class and then handle multiple different types of active blocks with random ticks, that would not be spam anymore then any other event, and in truth, with only so many actual random tick blocks, it shouldn't be as huge a problem as stated. (Or I'm pretty sure it wouldn't, unless every active block is a random tickable block and every player on a extremely massive server is spread far away, but then you'd probably already have performance issues anyhow)

**Comment by Actuarius** [comment]
@williewillus added labels [Feature]

**Pull request closed without merging**